### ISSUE TITLE
How to create wake word engine with AVS sdk

### ISSUE THREAD

**Issue by chesterkuo** [alexa/avs-device-sdk#470]
Hi There

I'm creating a open of wake word engine can be used in AVS system in following like.
https://github.com/chesterkuo/kaggle-speech-challenge-1
The Benchmark results of inference speed for speech (wav file) about 60-75ms and verify on RP3 platform.

I'd like to share and integrate into AVS so people can train this engine with their own speech data.
How to quick create a wake word engine for AVS SDK ?

There already example for python/C++ interface in github also.

**Comment by bandkenamazoncom** [comment]
Thanks for playing with this @chesterkuo!

You need to write your own interface.  There are two examples in the C++ SDK, KITT.ai and Sensory.  

At a high level, you would implement a class of type AbstractKeywordDetector, and in that class provide the plumbing to your engine.

Please look at the files under the KWD directory.

**Comment by anu-mathur** [comment]
Hi @chesterkuo  
Can we please talk more about your use case and experience with the AVS Device SDK? Please fill out this [Contact Form](https://www.amazon.com/forms/231ff404-261e-454c-917d-97208d9ea268) and I will get in touch with you.
Thanks

**Comment by chesterkuo** [comment]
@anu-mathur,

Form sent out.

**Comment by chesterkuo** [comment]
The model I had is based on CNN as network with mfcc input, and multiple label (pretrain model have "left", "right"...etc as spot word detection.


The frozen graph size about 5MB, and run time of inference is 60-75ms based on RP3.

People can train there own speech (word)by retraining to support your own data and apply as your spot word wake up engine.

For inference/test it , just use wav file as input and it will report prob result to see if it match spot words we had.

**Comment by celinval** [comment]
Hi @chesterkuo, have you tried to create an implementation of AbstractKeywordDetector as suggested above?

**Comment by chesterkuo** [comment]
@celinval ,

Not yet, as i'm working on another project to support ASR running on device site (without using cloud) for simple task , so for simple tasks like wether, alarm can be handled in local device instead of going to Cloud.

Please go ahead to handle it if you needs, and drop me if you need help from my side on Tensorflow inference of spot word detection, i'm happy to help on this.

chester.kuo@gmail.com

**Comment by mavamazon** [comment]
Hi Chester!

Thank you for your interest and effort! As for the original matter - do you still have questions on it?

**Comment by chesterkuo** [comment]
Maybe we can keep this open but use different label instead of closing it directly.

So people can still find this info if they need full source code implementation of wake up engine.

**Comment by kencecka** [comment]
Hi Chester,

This sounds like an interesting project.  I'm going to go ahead and close the issue for now since you don't have any specific problems waiting on feedback, but it will remain on github and be searchable.  Please feel free to reopen if you run into any problems with integration and we'll be glad to help out.

Ken
